### ISSUE TITLE
example or documents please

### ISSUE THREAD

**Issue by rojomisin** [jenkinsci/aws-credentials-plugin#11]
trying to figure out how to use this in a `Jenkinsfile`, finally got it to do something... 
```groovy
       withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: 'dev', variable: 'AWS_ACCESS_KEY_ID']]) {
               sh "echo this is ${env.AWS_ACCESS_KEY_ID}"
               sh "echo this is ${env.AWS_SECRET_ACCESS_KEY}"
       }
```
just displays **** but does not crash the script

Can someone give post a working hello world type code snippet example please? 
thanks!

**Comment by cbornet** [comment]
This is normal : withCredentials will hide the bound credentials from logs

**Comment by jfcarp** [comment]
Any example at all might make the plugin useful... zero example make it not useful.

**Comment by vedarthk** [comment]
I am sending these credentials to AWS CodeBuild project as environment variable, inside codebuild it should print in normal text but it prints `***`
I am writing them to a file in AWS CodeBuild which output `***` as well.

@cbornet is this a expected behaviour ?

**Comment by vedarthk** [comment]
I am sending these credentials to AWS CodeBuild project as environment variable, inside codebuild it should print in normal text but it prints `***`
I am writing them to a file in AWS CodeBuild which output `***` as well.

@cbornet is this a expected behaviour ?

**Comment by cbornet** [comment]
I don't know AWS CodeBuild, sorry...

**Comment by cbornet** [comment]
Not sure what the credential plugin is capable of hiding in the logs but I guess it should indeed be plain text here.

**Comment by burythehammer** [comment]
@vedarthk I believe `withCredentials` automatically hides the bound creds from any logs. That's to stop you, y'know, printing passwords/api secrets to your logs so anyone could get them.

**Comment by vedarthk** [comment]
@burythehammer you are right, it hides while showing it in console log in Jenkins. If I write it to file and download the artifact it outputs plain text in the file.

**Comment by cbornet** [comment]
It means that the plugin is able to catch the env shell command, nice !

**Comment by EArbeitman** [comment]
Was a resolution for this found? I'm trying to pass credentials from a declarative pipeline to an ANT script. But they seemed to be passing over as masked text. Thoughts?

**Comment by vedarthk** [comment]
@EArbeitman if you are looking at the console log, credentials will be masked in the logs. You should get them as plain text in your script and if you are writing them to console, you will see masked text.
